packages/graphql-toolbox/src/modules/SchemaView/Favorites/Favorites.tsx

```diff
@@ -0,0 +1,297 @@
+import { DragHandle } from "./DragHandle";
+import { FavoriteEntry } from "./FavoriteEntry";
+
+const screenReaderInstructions: ScreenReaderInstructions = {
+    draggable: `
+      To pick up a sortable item, press the space bar.
+      While sorting, use the arrow keys to move the item.
+      Press space again to drop the item in its new position, or press escape to cancel.
+    `,
+};
+
+interface FavoritesProps {
+    onSelectFavorite: (typeDefs: string) => void;
+}
+
+export const Favorites = ({ onSelectFavorite }: FavoritesProps) => {
+    const favorites = useStore((store) => store.favorites);
+    const selectedFavorites = useFavoritesStore((store) => store.selectedFavorites);
+    const [activeId, setActiveId] = useState<UniqueIdentifier | null>(null);
+    const [items, setItems] = useState<UniqueIdentifier[]>(favorites.map((favorite) => favorite.id));
+    const [showConfirm, setShowConfirm] = useState<boolean>(false);
+    const sensors = useSensors(
+        useSensor(PointerSensor),
+        useSensor(KeyboardSensor, {
+            coordinateGetter: sortableKeyboardCoordinates,
+        }),
+        useSensor(TouchSensor)
+    );
+
+    const isFirstAnnouncement = useRef(true);
+
+    const updateName = (newName: string, id: string): void => {
+        const updatedFavorites = favorites.map((fav) => (fav.id === id ? { ...fav, name: newName } : fav)) || null;
+        useStore.setState({ favorites: updatedFavorites });
+    };
+
+    const deleteSelectedFavorites = (): void => {
+        const remainingFavorites = favorites.filter((fav) => !selectedFavorites.includes(fav.id)) || null;
+        useFavoritesStore.setState({ selectedFavorites: [] });
+        useStore.setState({ favorites: remainingFavorites });
+    };
+
+    const downloadSelectedFavorites = (): void => {
+        const favoritesToDownload = favorites.filter((fav) => selectedFavorites.includes(fav.id));
+        favoritesToDownload.forEach((favorite) => {
+            const file = new Blob([favorite.typeDefs], {
+                type: "text/plain",
+            });
+            const downloadLink = document.createElement("a");
+            downloadLink.href = URL.createObjectURL(file);
+            downloadLink.download = `${encodeURI(favorite.name)}.txt`;
+            document.body.appendChild(downloadLink); // Required for this to work in FireFox
+            downloadLink.click();
+            downloadLink.remove();
+        });
+    };
+
+    const getIndex = (id: UniqueIdentifier) => items.indexOf(id);
+    const getPosition = (id: UniqueIdentifier) => getIndex(id) + 1;
+
+    useEffect(() => {
+        if (!activeId) {
+            isFirstAnnouncement.current = true;
+        }
+    }, [activeId]);
+
+    useEffect(() => {
+        if (favorites.length > items.length) {
+            setItems(favorites.map((favorite) => favorite.id));
+        }
+    }, [favorites]);
+
+    const announcements: Announcements = {
```

Comment:
Do you think it's worth moving the `announcements` here and the `screenReaderInstructions` above into a separate file? 
This file contains quite a few lines of code and maybe moving some of these more static/external things out could help to get a quicker grasp on what's going on. Wdyt?

Reply:
https://github.com/neo4j/graphql/pull/3506/commits/99931cdd7a0c7700072ae2fe931b6210228e6430